reactor-core/src/main/java/reactor/core/publisher/Sinks.java

```diff
@@ -186,6 +190,28 @@ public Emission getReason() {
 		}
 	}
 
+	/**
+	 * A handler for any non-successful emission result from operations
+	 * such as {@link Many#emitNext(Object, EmitFailureHandler)}
+	 * that allows retrying or failing the operation.
```

Comment:
I'd cover more ground here: give examples, state that this isn't a `Predicate` because side effects like waiting is allowed...